### ISSUE TITLE
Support for other CIs

### ISSUE THREAD

**Issue by asmeurer** [drdoctr/doctr#189]
In particular, GitLab and Circle are popular.

**Comment by gforsyth** [comment]
Info for encrypted variables in Circle: https://github.com/circleci/encrypted-files

**Comment by asmeurer** [comment]
If any of them let us upload a key directly we should do that.

**Comment by gforsyth** [comment]
With circle we would still be uploading the key to github, no?  gitlab is a different beast.

**Comment by asmeurer** [comment]
I mean getting the key onto the CI. Everything would still use gh-pages. I don't know if Circle or GitLab support uploading the key directly, so that we don't have to encrypt it ourselves. Travis supports this, but only for private accounts (c.f. https://github.com/travis-ci/travis-ci/issues/7420).

**Comment by asmeurer** [comment]
[This page](https://circleci.com/docs/1.0/github-security-ssh-keys/) claims that a deploy key is added automatically. I'm guessing they are set as read-only, so I don't know if that can be changed.  If not, [this page](https://circleci.com/docs/1.0/permissions-and-access-during-deployment/) claims that you can add ssh keys.

It looks like [GitLab CI](https://docs.gitlab.com/ee/ci/ssh_keys/README.html) requires that we do basically the same thing that we do on Travis.

**Comment by CJ-Wright** [comment]
This is awesome! We use GitLab a bit so it would be great to have this!

**Comment by asmeurer** [comment]
I think [this](https://docs.gitlab.com/ce/api/build_variables.html#create-variable) is the relevant API call for GitLab CI.

**Comment by asmeurer** [comment]
GitLab is hard just because, as far as I can tell, you can't actually integrate GitLab CI with GitHub. That means we could still implement the feature, but we would be unable to test it for this repo.

**Comment by CJ-Wright** [comment]
Does the project need a gitlab repo for testing with gitlabCI?

**Comment by asmeurer** [comment]
We can set up a GitLab repo easily enough. The problem is that we'd like to have continuous testing of doctr itself, so that whenever we make changes to doctr, we can be reasonably sure that the GitLab stuff isn't breaking.

**Comment by asmeurer** [comment]
I made a GitLab group https://gitlab.com/doctr

**Comment by CJ-Wright** [comment]
Can you run the CI on gitlab and then look at the badge before accepting the PR? It would be nice if there was a PR feature looked Travis that showed of the gitlab build passed.

**Comment by asmeurer** [comment]
Yeah as far as I saw there's no way to use GitLab CI as GitHub status checks.

**Comment by asmeurer** [comment]
For Azure Pipelines: https://github.com/Microsoft/azure-pipelines-tasks/issues/9172#issuecomment-466024682

**Comment by asmeurer** [comment]
Also https://github.com/drdoctr/doctr/issues/346
